libs/types/src/tokens.rs

```diff
@@ -322,17 +375,114 @@ mod tests {
 			))
 			.is_err()
 		);
-		assert!(
-			TryInto::<GeneralCurrencyIndex<u128, ZeroPrefix>>::try_into(CurrencyId::KSM).is_err()
-		);
-		assert!(
-			TryInto::<GeneralCurrencyIndex<u128, ZeroPrefix>>::try_into(CurrencyId::AUSD).is_err()
-		);
 		assert!(
 			TryInto::<GeneralCurrencyIndex<u128, ZeroPrefix>>::try_into(CurrencyId::Staking(
 				StakingCurrency::BlockRewards
 			))
 			.is_err()
 		);
 	}
+
+	#[cfg(test)]
+	mod tests {
+		use cfg_primitives::TrancheId;
+		use codec::Encode;
+		use hex::FromHex;
+
+		use super::StakingCurrency;
+		use crate::{tokens as after, tokens::before};
+
+		#[test]
+		fn currency_id_refactor_encode_equality() {
+			// Native
+			assert_eq!(
+				before::CurrencyId::Native.encode(),
+				after::CurrencyId::Native.encode()
+			);
+			assert_eq!(after::CurrencyId::Native.encode(), vec![0]);
+
+			// Tranche
+			assert_eq!(
+				before::CurrencyId::Tranche(33, default_tranche_id()).encode(),
+				after::CurrencyId::Tranche(33, default_tranche_id()).encode()
+			);
+			assert_eq!(
+				after::CurrencyId::Tranche(33, default_tranche_id()).encode(),
```

Comment:
Do we plan on keeping before forever then?

Reply:
No! We can even drop this now, I just wrote this test to ensure that the different variants would encode accordingly to the new structure. Wdyt?

Reply:
Sounds good! Thanks for the explainer

Reply:
Maybe let's remove in the follow up clean up pr. Makes the pr clearer for future us. Wdyt?

Reply:
Yeah let's do that ✌🏻 thanks for the suggestion!